### PR TITLE
[doc] Add more explicit migration notes.

### PR DESCRIPTION
**Pull request by wsargent**
* Adds details about i18nsupport migration needing an implicit request.
* Adds details of using `MessagesRequest` in migration.

### CODE REVIEW

File: documentation/manual/releases/release26/migration26/MessagesMigration26.md [hunk 1]
```diff
@@ -22,9 +22,45 @@ The static deprecated methods in [`play.i18n.Messages`](api/java/play/i18n/Messa
...
 [`Messages`](api/scala/play/api/i18n/Messages.html) is now a trait (rather than a case class).  The case class is now [`MessagesImpl`](api/scala/play/api/i18n/MessagesImpl.html), which implements [`Messages`](api/scala/play/api/i18n/Messages.html).
 
+### I18nSupport Implicit Conversion
+
+If you are upgrading directly from Play 2.4 to Play 2.5, you should know that `I18nSupport` support changed in 2.5.x.  In 2.5.x, it was possible through a series of implicits to use a "language default" `Messages` instance if the request was not declared to be in implicit scope:
+
+```scala
+  def listWidgets = Action {
+    val lang = implicitly[Lang] // Uses Lang.defaultLang
+    val messages = implicitly[Messages] // Uses I18NSupport.lang2messages
```
Comment (gmethvin) [inline documentation/manual/releases/release26/migration26/MessagesMigration26.md]: `Uses I18nSupport.lang2messages(Lang.defaultLang)`

File: documentation/manual/releases/release26/migration26/MessagesMigration26.md [hunk 2]
```diff
@@ -22,9 +22,45 @@ The static deprecated methods in [`play.i18n.Messages`](api/java/play/i18n/Messa
 
 [`Messages`](api/scala/play/api/i18n/Messages.html) is now a trait (rather than a case class).  The case class is now [`MessagesImpl`](api/scala/play/api/i18n/MessagesImpl.html), which implements [`Messages`](api/scala/play/api/i18n/Messages.html).
 
+### I18nSupport Implicit Conversion
+
+If you are upgrading directly from Play 2.4 to Play 2.5, you should know that `I18nSupport` support changed in 2.5.x.  In 2.5.x, it was possible through a series of implicits to use a "language default" `Messages` instance if the request was not declared to be in implicit scope:
```
Comment (wsargent) [inline documentation/manual/releases/release26/migration26/MessagesMigration26.md]: "Prior to 2.6.x" is probably better wording.

File: documentation/manual/releases/release26/migration26/MessagesMigration26.md [hunk 3]
```diff
@@ -22,9 +22,45 @@ The static deprecated methods in [`play.i18n.Messages`](api/java/play/i18n/Messa
...
+    val messages = implicitly[Messages] // Uses I18NSupport.lang2messages
+    // implicit parameter messages: Messages in requiresMessages template, but no request!
+    val content = views.html.requiresMessages(form)
+    Ok(content)
+  }
+```
+
+The [`I18nSupport`](api/scala/play/api/i18n/I18nSupport.html) implicit conversion now requires an implicit request or request header in scope in order to correctly determine the preferred locale and language for the request.
+
+> **If you are seeing an error message like `could not find implicit value for parameter messages: play.api.i18n.Messages` then you need to explicitly add the request as an implicit.**
```
Comment (richdougherty) [inline documentation/manual/releases/release26/migration26/MessagesMigration26.md]: Could you add an `@implicitNotFound` annotation to `Messages` so the user gets a helpful compiler error?

Comment (wsargent) [inline documentation/manual/releases/release26/migration26/MessagesMigration26.md]: I will do that.

**Pull request merged**